src/main/java/racing/model/Cars.java

```diff
@@ -1,32 +1,54 @@
 package racing.model;
 
-import java.util.Collections;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Random;
 import java.util.stream.Collectors;
-import racing.view.ResultView;
 
 public class Cars {
+    private final Random random = new Random();
+    public static final int START_PROGRESS = 0;
     private final List<RacingCar> cars;
 
-    public Cars(List<RacingCar> cars) {
-        this.cars = Collections.unmodifiableList(cars);
+    Cars(List<RacingCar> cars) {
+        this.cars = cars;
     }
-    public void raceAllCars() {
-        for (RacingCar car : cars) {
-            car.race(car.getRandomInt());
-            ResultView.showRaceProgress(car);
-        }
-    }
-    public static List<RacingCar> getWinners(List<RacingCar> cars) {
+
+    public List<String> getWinnerNames() {
         return cars.stream()
-                .filter(car -> car.isSameProgress(getTopSpeed(cars)))
+                .filter(car -> car.isSameProgress(getTopSpeed()))
+                .map(RacingCar::getName)
                 .collect(Collectors.toList());
     }
-    public static int getTopSpeed(List<RacingCar> cars) {
+
+    public List<RacingCar> cars() {
+        return this.cars;
+    }
+
+    public void raceAllCars(){
```

Comment:
`raceAllCars`에 대한 테스트를 하기 힘든구조 같아요 ㅜ
왜냐하면 내부 로직에 `random.nextInt(10)`이 고정되어 있기 때문에 그렇게 보여지네요 🤔 

```suggestion
    public void raceAllCars(Generator generator){
```

숫자 생성기를 주입받아보는 관점으로 접근해보면 어떨까요?
그렇다면, 테스트 입장에서 전달하고자 하는 객체를 전달할수 있을것 같아서요!

`RandomGenerator`, `ManualGenerator` 이런식을 도출 해낼수 있을것 같아요 🤔

Reply:
<img width="963" alt="스크린샷 2023-11-18 오후 4 12 10" src="https://github.com/next-step/java-racingcar/assets/12059592/2cd4da71-7671-4256-a063-ced4df96fa83">